app/models/station.rb

```diff
@@ -1,3 +1,71 @@
 class Station < ActiveRecord::Base
-  validates :name, :dock_count, :city, :installation_date, presence: true
+  belongs_to :city
+  validates :name,
+            :dock_count,
+            :city_id,
+            :installation_date,
+            presence: true
+
+  def self.record(station_information)
+    self.find_or_create_by(name:       station_information[:name],
+                           dock_count: station_information[:dock_count],
+                           city_id: find_city_id(station_information[:city_name]),
+                           installation_date: station_information[:installation_date]
+                          )
+  end
+
+  def self.record_update(station, station_information)
+    self.find_or_create_by(name:       station_information[:name],
+                           dock_count: station_information[:dock_count],
+                           city_id: find_city_id(station_information[:city_name]),
+                           installation_date: station_information[:installation_date]
+                          )
+  end
+
+  def find_city_id(city_name)
```

Comment:
Honestly, i'm not sure if we need both of these methods. (the one with/without self) probably going to ask someone about it tomorrow!